### PR TITLE
Topic deduplication options, Halyard, and Go v1.20

### PR DESCRIPTION
**Pull request by bbengfort**
### Scope of changes

Updates the topic protocol buffers with deduplication options and adds a simple migration to update the topics with the None deduplication strategy.

Fixes SC-20990

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

This is the first step towards event deduplication. Do we have enough information stored on the topic to make deduplication possible?

### Definition of Done

- [ ] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [x] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [x] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [x] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [x] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: proto/api/v1beta1/event.proto [hunk 1]
```diff
@@ -41,6 +41,14 @@ message EventWrapper {
     Encryption encryption = 10;
     Compression compression = 11;
 
+    // Deduplication pointer; if this event wrapper is a duplicate then it should be
+    // omitted from standard queries depending on the deduplication strategy of the
+    // topic. The DuplicateID points to the event that contains the data referred to by
+    // this duplicated event wrapper. A duplicate event wrapper will likely have all of
+    // the metadata but no event data.
+    bool is_duplicate = 12;
+    bytes duplicate_id = 13;
```
Comment (pdeziel) [inline proto/api/v1beta1/event.proto:50]: Sounds like we will need to rebuild the protocol buffers for the SDKs?

**Review by pdeziel — APPROVED** [review]
Looks good, I'm sure that utility will come in handy.

File: proto/api/v1beta1/event.proto (hunk 1 above)
Comment (bbengfort) [reply proto/api/v1beta1/event.proto:50]: Yes, but not urgent since we're adding protobuf fields -- so it should be backwards compatible.

**Comment by bbengfort** [comment]
CI is erroring because `errors.Join` got added in go 1.20 … so I'm going to update us to go 1.20.

**Pull request merged**